extraction_tools_wiki/extract_wiki_data.py

```diff
@@ -28,106 +28,116 @@
+    user_agent = "some-agent"
+    user_email = "name@domain.com"
 
     # Boolean to trigger load page titles from file, or run fresh page title extraction
-    LOAD_FILES = False
+    load_files = True
 
     # Set the revision date, extract wiki pages only after this date
-    LAST_EXTRACTION_DATE = datetime.datetime.strptime("2019-01-29T00:00:00Z",
+    last_extraction_date = datetime.datetime.strptime("2019-01-28T00:00:00Z",
                                                       '%Y-%m-%dT%H:%M:%SZ')
 
     # STAGE ONE: EXTRACT PAGE TITLES
 
     print(">>> Starting wiki page titles extraction...")
     # Create object to handle page titles extraction
-    WIKI_PAGE_TITLES = WikiPageTitles(TARGET_CATEGORIES,
-                                      TITLES_FILE_PATH,
-                                      USER_AGENT,
-                                      USER_EMAIL)
+    wiki_page_titles = WikiPageTitles(OSRS_WIKI_API_URL,
+                                      target_categories,
+                                      user_agent,
+                                      user_email)
 
     # Load previously extracted page titles from JSON, or extract from OSRS Wiki API
-    if LOAD_FILES:
-        LOADED_PAGE_TITLES = WIKI_PAGE_TITLES.load_page_titles()
-        if not LOADED_PAGE_TITLES:
-            print(">>> ERROR: Specified page titles to load, but not file found...")
-            print(">>> EXITING.")
-            sys.exit(1)
+    if load_files:
+        loaded_page_titles = wiki_page_titles.load_page_titles(titles_file_path)
+        if not loaded_page_titles:
+            sys.exit(">>> ERROR: Specified page titles to load, but not file found. Exiting.")
     else:
-        WIKI_PAGE_TITLES.extract_page_titles()
-        WIKI_PAGE_TITLES.extract_last_revision_timestamp()
-        WIKI_PAGE_TITLES.export_page_titles_in_json()
+        # Extract page titles using supplied categories
+        wiki_page_titles.extract_page_titles()
+        # Extract page revision date
+        # Loop 50 page titles at a time, the max number for a revisions request using page titles
+        for page_title_list in itertools.zip_longest(*[iter(wiki_page_titles.page_titles)] * 50):
+            # Remove None entries from the list of page titles
+            page_title_list = filter(None, page_title_list)
+            # Join the page titles list using the pipe (|) separator
+            page_titles_string = "|".join(page_title_list)
+            # Extract the page revision date
+            wiki_page_titles.extract_last_revision_timestamp(page_titles_string)
+        # Save all page titles and
+        wiki_page_titles.export_page_titles_in_json(titles_file_path)
 
     # Determine page titles count
-    PAGE_TITLES_TOTAL = len(WIKI_PAGE_TITLES.page_titles)
-    print(">>> Number of extracted wiki pages: %d" % PAGE_TITLES_TOTAL)
+    page_titles_total = len(wiki_page_titles)
+    print(">>> Number of extracted wiki pages: %d" % page_titles_total)

```

Comment:
Would use f-strings here as well. It will automatically format different variables into their string type so you don't always have to remember the `%d`, `%r` etc